### ISSUE TITLE
Add an option to install "Minimal Racket"

### ISSUE THREAD

**Issue by samth** [greghendershott/travis-racket#4]
This would probably be "racket-textual" for versions 5.3.6 and before.

**Comment by Dashed** [comment]
Pretty easy to amend. I made the appropriate changes at https://github.com/Dashed/travis-racket

**Comment by samth** [comment]
@Dashed can you turn that into a pull request? That would probably make it easiest for @greghendershott.

**Comment by greghendershott** [comment]
@samth Any preference on how the user should specify this in `.travis.yml`? Some options:

1. `RACKET_VERSION` values can now be something like `6.1.1/minimal` as well as `6.1.1`.

2. A new variable (named something like `RACKET_EDITION` or `RACKET_SIZE`) has values like `minimal`, `full`.

I think 2 is cleanest, assuming the Travis matrix works the way I recall. That is if you specify:

```
matrix:
  - RACKET_VERSION = 1
  - RACKET_VERSION = 2
  - RACKET_VERSION = 3
  - RACKET_EDITION = full
  - RACKET_EDITION = minimal
```

Then you would get 6 builds, full and minimal each of 1, 2, 3.

Does that make sense? Any other and/or better ideas?

**Comment by samth** [comment]
@greghendershott option 2 looks nicest to me, too.

**Comment by greghendershott** [comment]
OK, good. (Although it wouldn't let someone say to build, say, (a) version 1 full and (b) version 2 minimal -- and only those -- it's hard for me imagine that being a real need.)

I have another PR to merge today, I'll try to loop back after that and do these changes after.

**Comment by samth** [comment]
No, you can even do that: http://docs.travis-ci.com/user/environment-variables/#Matrix-Variables

**Comment by greghendershott** [comment]
Ah, that's right. Perfect.

**Comment by greghendershott** [comment]
So as I work on this, I realize that Minimal Racket is _so_ minimal that `raco` has only `setup` and `pkg` subcommands. Not even `make` or `test`.

I'll have it helpfully do a `raco pkg install rackunit`. But, that's tricky to get right both for 6.0 Minimal Racket and for 5.3.x "textual".  And the more I think about it, this should probably just support Minimal Racket, for 6.0 and HEAD (snapshots). So I think I'm going to focus it.

**Comment by greghendershott** [comment]
So, it works but the total build times are noticeably longer for Minimal Racket. The savings of downloading 1/10th are more than offset by `raco pkg install rackunit` -- rackunit and all its deps.  And if the user's code needs even more packages that would have been in the full distribution....

I guess I could try --no-setup and a final `raco setup -D`, to avoid building the docs, for rackunit. And I guess README that people should do similar for their own code?

**Comment by greghendershott** [comment]
Example timings: https://travis-ci.org/greghendershott/travis-racket/builds/63104516

Also, the Minimal HEAD permutation is fragile. That Northwestern catalog server failed a job do to a TCP reset. I restarted that job. As I type this, it's been stuck for 5 minutes on "Resolving "rackunit" via http://plt.eecs.northwestern.edu/snapshots/20150518-542b960/catalog/". Ugh.

**Comment by samth** [comment]
Just installing "rackunit-lib" will save a lot. Also, `raco test` is in the "compiler-lib" pkg.

**Comment by greghendershott** [comment]
1. Thanks, I'm trying `rackunit-lib` now.

2. That Northwestern catalog server isn't looking good:

```
Minimal Racket: Installing rackunit package...

Resolving "rackunit" via http://plt.eecs.northwestern.edu/snapshots/20150518-542b960/catalog/

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated
```

**Comment by greghendershott** [comment]
@samth Unfortunately it looks like `rackunit-lib` does not add a `raco test` subcommand -- https://travis-ci.org/greghendershott/travis-racket/jobs/63106407 -- so will need to switch back to `rackunit`.

**Comment by samth** [comment]
Like I said, that's in "compiler - lib".

**Comment by greghendershott** [comment]
Oh, I misunderstood you to say it was from rackunit-lib, and also from compiler-lib.

IIUC now I want to try rackunit-lib and compiler-lib, both.
